docs/modules/under_the_hood.md

```diff
@@ -0,0 +1,46 @@
+# Isaac Sim Details
+In this page, we discuss the particulars of certain Isaac Sim features and behaviors.
+
+## Playing and Stopping
+TODO
+
+## CPU and GPU dynamics and pipelines
+TODO
+
+## Sources of Truth: USD, PhysX and Fabric
```

Comment:
could be good to anchor each section in terms of pros / cons, to motivate why all three are concurrently used. e.g.: USD: slow, but complete, Fabric: fast, but constrained, etc. The descriptions do a good job of explaining it, but it could be helpful to have a concise table with only a few words describing pros / cons for people to intuitively get a feel for each of them